lib/Analysis/Utility.cpp

```diff
@@ -633,14 +633,39 @@ std::unique_ptr<DataFlowSolver> createDataFlowSolver() {
   return solver;
 }
 
-static triton::MakeTensorPtrOp getMakeTensorPtrOpImpl(Operation *op, Value v) {
+struct BlockToCFOpsHelper {
+  BlockToCFOpsHelper(Value v) {
+    auto moduleOp =
+        v.getParentBlock()->getParentOp()->getParentOfType<ModuleOp>();
+    moduleOp.walk([&](Operation *op) {
+      if (auto br = dyn_cast<cf::BranchOp>(op)) {
+        Block *block = br.getDest();
+        blockToCFOps[block].insert({op, -1});
+      }
+      if (auto condBr = dyn_cast<cf::CondBranchOp>(op)) {
+        Block *blockT = condBr.getTrueDest();
+        Block *blockF = condBr.getFalseDest();
+        blockToCFOps[blockT].insert({condBr, 1});
+        blockToCFOps[blockF].insert({condBr, 0});
+      }
+    });
+  }
+
+  using BranchOps = llvm::SetVector<std::pair<Operation *, int>>;
+  llvm::DenseMap<Block *, BranchOps> blockToCFOps;
+};
+
+static triton::MakeTensorPtrOp getMakeTensorPtrOp_(BlockToCFOpsHelper &helper,
```

Comment:
`getMakeTensorPtrOp_` -> using `_` at the end doesn't follow the naming convention?

Reply:
I will change.